### PR TITLE
ZA2-Harerimana-Dieudonne-JavaScript-Core-1-Coursework-Week1

### PR DESCRIPTION
**Pull request by diegoharera**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Review by BereniceOgundele — COMMENTED** [review]
Hi Harerimana, it looks like you put in a lot of time and effort to complete these exercises. Well done for hanging in there and completing it. All the best going forward...your code looks good.

File: mandatory/4-tax.js [hunk 1]
```diff
@@ -16,9 +18,9 @@ function calculateSalesTax() {}
 
   Remember that the prices must include the sales tax (hint: you already wrote a function for this!)
 */
-
-function addTaxAndFormatCurrency() {}
-
+function addTaxAndFormatCurrency(number) {
+  return "£" + calculateSalesTax(number).toFixed(2);
+}
 /*
```
Comment (lexyking) [inline mandatory/4-tax.js:24]: Great job on these functions, Can you please have a look at the other function ??

File: mandatory/3-function-output.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+// The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.
 function combine2Words(word1, word2) {
   return word1.concat(word2);
 }
 
 function concatenate(firstWord, secondWord, thirdWord) {
   // Write the body of this function to concatenate three words together.
   // Look at the test case below to understand what this function is expected to return.
+  return firstWord.concat(" ", secondWord, " ", thirdWord);
 }
```
Comment (lexyking) [inline mandatory/3-function-output.js:17]: Amazing.

File: mandatory/2-logic-error.js [hunk 3]
```diff
@@ -1,16 +1,13 @@
 // The syntax for this function is valid but it has an error, find it and fix it.
 
 function trimWord(word) {
-  return wordtrim();
+  return word.trim();
 }
-
 function getStringLength(word) {
-  return "word".length();
+  return word.length();
```
Comment (lexyking) [inline mandatory/2-logic-error.js:7]: `length` is not called with the `()`

File: mandatory/1-syntax-errors.js [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
-function addNumbers(a b c) {
+function addNumbers(a, b, c) {
   return a + b + c;
 }
-
-function introduceMe(name, age)
-  return "Hello, my name is " + name "and I am " age + "years old";
-
+function introduceMe(name, age) {
+  return "Hello, my name is " + name + " and I am " + age + " years old";
```
Comment (lexyking) [inline mandatory/1-syntax-errors.js:7]: Well done on this one, I like your attention to detail on those empty spaces.

File: exercises/L-functions-nested/exercise.js [hunk 5]
```diff
@@ -3,3 +3,12 @@ var mentor2 = "Irina";
...
 var mentor5 = "Yohannes";
+function greet(name) {
+  let greeting = "Hello " + name;
+  return greeting.toLocaleUpperCase();
+}
+console.log(greet(mentor1));
+console.log(greet(mentor2));
+console.log(greet(mentor3));
+console.log(greet(mentor4));
+console.log(greet(mentor5));
```
Comment (lexyking) [inline exercises/L-functions-nested/exercise.js:14]: Well done

File: exercises/K-functions-parameters/exercise.js [hunk 6]
```diff
@@ -1,9 +1,8 @@
 // Complete the function so that it takes input parameters
-function multiply() {
+function multiply(a, b) {
   // Calculate the result of the function and return it
+  return a * b;
 }
-
 // Assign the result of calling the function the variable `result`
 var result = multiply(3, 4);
-
```
Comment (lexyking) [inline exercises/K-functions-parameters/exercise.js:8]: Well done.

File: exercises/K-functions-parameters/exercise2.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 // Declare your function first
-
-var result = divide(3, 4);
-
+function divide(a, b) {
+  return a / b;
+}
+var result = divide(1.5, 2);
```
Comment (lexyking) [inline exercises/K-functions-parameters/exercise2.js:5]: Well done

File: exercises/K-functions-parameters/exercise3.js [hunk 8]
```diff
@@ -1,5 +1,6 @@
 // Write your function here
-
-var greeting = createGreeting("Daniel");
-
+function createGreeting(name) {
+  return "Hello my name is " + name;
+}
+var greeting = createGreeting("Diego");
```
Comment (lexyking) [inline exercises/K-functions-parameters/exercise3.js:5]: Well done

File: exercises/K-functions-parameters/exercise4.js [hunk 9]
```diff
@@ -1,5 +1,7 @@
 // Declare your function first
-
+function add(a, b) {
+  return a + b;
+}
 // Call the function and assign to a variable `sum`
-
+let sum = add(13, 124);
 console.log(sum);
```
Comment (lexyking) [inline exercises/K-functions-parameters/exercise4.js:7]: Nice

File: exercises/K-functions-parameters/exercise5.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
-// Declare your function here
-
-const greeting = createLongGreeting("Daniel", 30);
-
+function createLongGreeting(name, age) {
+  return "Hello my name is " + name + " and I'm " + age + " years old";
+}
+const greeting = createLongGreeting("Diego", 30);
```
Comment (lexyking) [inline exercises/K-functions-parameters/exercise5.js:4]: Well done

File: exercises/J-functions/exercise2.js [hunk 11]
```diff
@@ -1,5 +1,7 @@
 function triple(number) {
   // complete function here
+
+  return number * 3;
 }
```
Comment (lexyking) [inline exercises/J-functions/exercise2.js:5]: Nice

File: exercises/J-functions/exercise.js [hunk 12]
```diff
@@ -1,5 +1,6 @@
 function halve(number) {
   // complete the function here
+  return number / 2
 }
```
Comment (lexyking) [inline exercises/J-functions/exercise.js:5]: Nice

File: exercises/I-floats/exercise.js [hunk 13]
```diff
@@ -1,2 +1,9 @@
 var numberOfStudents = 15;
 var numberOfMentors = 8;
+
+let studentPercentage = (numberOfStudents / 23) * 100;
+let mentorsPercentage = (numberOfMentors / 23) * 100;
+let studentsResult = Math.round(studentPercentage);
+let mentorsResult = Math.round(mentorsPercentage);
+console.log("Percentage students:", studentsResult + "%");
+console.log("Percentage students:", mentorsResult + "%");
```
Comment (lexyking) [inline exercises/I-floats/exercise.js:9]: Great job!

File: exercises/G-numbers/exercise.js [hunk 14]
```diff
@@ -1 +1,7 @@
 // Start by creating a variables `numberOfStudents` and `numberOfMentors`
+let numberOfStudents = 15;
+let numberOfMentors = 8;
+let total = numberOfStudents + numberOfMentors;
+console.log("NumberOfStudents:", numberOfStudents);
+console.log("let numberOfMentors:", numberOfMentors);
+console.log("Total number of students and mentors:", total);
```
Comment (lexyking) [inline exercises/G-numbers/exercise.js:7]: Well done.

File: exercises/F-strings-methods/exercise2.js [hunk 15]
```diff
@@ -1,3 +1,3 @@
 const name = " Daniel  ";
-
+const message = name.trim();
```
Comment (lexyking) [inline exercises/F-strings-methods/exercise2.js:2]: Good use of the `.trim` method but the ask was to rewrite the same message as above.

File: exercises/F-strings-methods/exercise.js [hunk 16]
```diff
@@ -1,3 +1,5 @@
 // Start by creating a variable `message`
-
+let name = "Diego";
+let amount = name.length;
+let message = "hello,my name is " + name + " and my name is " + amount + " characters long";
```
Comment (lexyking) [inline exercises/F-strings-methods/exercise.js:4]: Well done.

File: exercises/E-strings-concatenation/exercise.js [hunk 17]
```diff
@@ -1,3 +1,6 @@
 // Start by creating a variable `message`
+let greetingStart = "hello,my name is ";
+let name = "DIego";
 
-console.log(message);
+let greeting = greetingStart + name;
+console.log(greeting);
```
Comment (lexyking) [inline exercises/E-strings-concatenation/exercise.js:6]: Great.

File: exercises/D-strings/exercise.js [hunk 18]
```diff
@@ -1,3 +1,4 @@
 // Start by creating a variable `message`
-
+let message = "my name is Harera";
 console.log(message);
+console.log(typeof message);
```
Comment (lexyking) [inline exercises/D-strings/exercise.js:4]: Well done.

File: exercises/C-variables/exercise.js [hunk 19]
```diff
@@ -1,3 +1,6 @@
 // Start by creating a variable `greeting`
-
+let greeting = "hello Africa";
+console.log(greeting);
+console.log(greeting);
+console.log(greeting);
```
Comment (lexyking) [inline exercises/C-variables/exercise.js:5]: Well done.

File: exercises/B-hello-world/exercise.js [hunk 20]
```diff
@@ -1 +1,5 @@
-console.log("Hello world");
+// console.log("Hello world");
+//console.log("save the world");
+// console.log("create the world","happy","code your");
+// console.log(save the world);
+console.log(39);
```
Comment (lexyking) [inline exercises/B-hello-world/exercise.js:5]: Well done.

**Review by lexyking — CHANGES_REQUESTED** [review]
Left few comments but overall great jobs.

**Pull request closed without merging**